flink-streaming-java/src/main/java/org/apache/flink/streaming/api/environment/StreamExecutionEnvironment.java

```diff
@@ -2081,6 +2101,20 @@ public JobClient executeAsync(StreamGraph streamGraph) throws Exception {
                 configuration.get(DeploymentOptions.TARGET),
                 "No execution.target specified in your configuration file.");
 
+        errors.addAll(checkNotAllowedConfigurations());
+        if (!errors.isEmpty()) {
+            // HACK: We shortcut the StreamGraph to jobgraph translation because we already
+            // know that the job needs to fail and can derive the jobId.
+
+            final JobID jobId =
+                    configuration
+                            .getOptional(PipelineOptionsInternal.PIPELINE_FIXED_JOB_ID)
+                            .map(JobID::fromHexString)
+                            .orElse(new JobID());
```

Comment:
Can we factor this out into a common utility to avoid duplication?

Reply:
I checked all production cases accessing the `PIPELINE_FIXED_JOB_ID` field and they all handle the return value of the optional differently. Unfortunately, I do not see an easy way to duplicate things without making deeper changes to when and where we generate the jobId.